packages/jsonc/jsonc.mjs

```diff
@@ -0,0 +1,36 @@
+import { getFilenameExt } from '@nodejs-loaders/parse-filename';
+import stripJsonComments from 'strip-json-comments';
+
+/**
+ * @type {import('node:module').ResolveHook}
+ */
+async function resolveJSONc(specifier, ctx, nextResolve) {
+	const nextResult = await nextResolve(specifier);
+	const ext = getFilenameExt(nextResult.url);
+
+	if (ext === '.jsonc' && ctx.importAttributes.type === 'jsonc')
+		return {
+			...nextResult,
+			format: 'jsonc'
+		};
+
+	return nextResult;
+}
+export { resolveJSONc as resolve }
+
+/**
+ * @type {import('node:module').LoadHook}
+ */
+async function loadJSONc(url, ctx, nextLoad) {
+	if (ctx.format !== 'jsonc') return nextLoad(url);
+	const nextResult = await nextLoad(url, ctx);
```

Comment:
nit: You're running `nextLoad` regardless, so I would just cache the result before checking, and then early return that cached result. A bit DRY'er and less complicated code branching :)

```suggestion
	const nextResult = await nextLoad(url);
	
	if (ctx.format !== 'jsonc') return nextResult;
```